litgpt/eval/evaluate.py

```diff
@@ -36,10 +36,10 @@ def prepare_results(results, save_filepath, print_results=True):
 
 
 def convert_and_evaluate(
-    checkpoint_dir: str,
+    checkpoint_dir: Optional[str] = None,
```

Comment:
Why is this optional now?

Reply:
It wasn't clear to me how to capture whether an argument wasn't provided in jsonargparse. Basically without it you get an

```
⚡ print-task-list ~/litgpt litgpt evaluate
usage: litgpt [-h] [-c CONFIG] [--print_config[=flags]] {download,chat,finetune,pretrain,generate,convert,merge_lora,evaluate} ...
error: Validation failed: Key "evaluate.checkpoint_dir" is required but not included in config object or its value is None.
```

How to best capture that without a general `try/except`?

Reply:
We do the same thing in `litgpt download` with `repo_id`, which is also optional. Unless there are strong concerns or there is a better way to do it, I'd say we keep it and merge. What do you think?

Reply:
In download.py, if you don't pass the repo_id, it raises an error giving you concrete instructions about it.

On the other hand, here if you don't pass a `checkpoint_dir`, you'll get a generic exception when it calls `Path(checkpoint_dir)`.

If you want to list the tasks, you could still do that without making `checkpoint_dir` optional.

They are unrelated arguments. One could also make the argument that `litgpt evaluate` should list the available `checkpoint_dir`s that are found locally. Or the checkpoints and tasks.

What's inconvenient about having the `checkpoint_dir` still be required to show the tasks? It will need to be given anyways after the user decides what task(s) to select

Reply:
> What's inconvenient about having the checkpoint_dir still be required to show the tasks? It will need to be given anyways after the user decides what task(s) to select

I was thinking that it creates unnecessary typing effort. I.e., if you just quickly want to know the tasks, you have to type the `--checkpoint_dir ...` even though it's from a technical side not involved in printing this task list.

Happy to modify it though; it's not a real major inconvenience.

The updated version should have it now.